### PR TITLE
feat(account-lib): cardano ada coin skeleton

### PR DESCRIPTION
**Pull request by margueriteblair**
Ticket: BG-51368

### CODE REVIEW

File: modules/account-lib/src/coin/ada/index.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import * as Utils from './utils';
```
Comment (ewangbitgo) [inline modules/account-lib/src/coin/ada/index.ts:1]: i thought we are creating a separate module for ada, i.e. sdk-core-ada, not in account-lib

**Comment by mmcshinsky-bitgo** [comment]
Looks like this was done manually into a new coin. To make your life easier, we do a lot of automation here: https://github.com/BitGo/BitGoJS/blob/master/DEVELOPERS.md#new-coin-implementation

File: modules/sdk-coin-ada/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+In your project install both `@bitgo/sdk-api` and `@bitgo/sdk-coin-ada`.
+
+```shell
+npm i @bitgo/sdk-api @bitgo/sdk-coin-ada
+```
+
+Next, you will be able to initialize an instance of "bitgo" through `@bitgo/sdk-api` instead of via `bitgo`.
+
+```javascript
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-ada/README.md:17]:
```suggestion
```javascript
import { BitGoAPI } from '@bitgo/sdk-api';
import { Ada } from '@bitgo/sdk-coin-ada';

const sdk = new BitGoAPI();

sdk.register('ada', Ada.createInstance);
```

File: modules/sdk-coin-ada/src/keyPair.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import { DefaultKeys, KeyPairOptions, Ed25519KeyPair, AddressFormat } from 'modules/sdk-core/src';
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-ada/src/keyPair.ts:1]:
```suggestion
import { DefaultKeys, KeyPairOptions, Ed25519KeyPair, AddressFormat } from '@bitgo/sdk-core';
```

File: modules/sdk-coin-ada/src/transaction.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { BaseTransaction, BaseKey, NotImplementedError } from 'modules/sdk-core/src';
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-ada/src/transaction.ts:1]:
```suggestion
import { BaseTransaction, BaseKey, NotImplementedError } from '@bitgo/sdk-core';
```

File: modules/sdk-coin-ada/src/transactionBuilder.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
+import BigNumber from 'bignumber.js';
+import { BaseCoin as CoinConfig } from 'modules/statics/src/base';
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-ada/src/transactionBuilder.ts:2]:
```suggestion
import { BaseCoin as CoinConfig } from '@bitgo/statics';
```

File: modules/sdk-coin-ada/src/transactionBuilder.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+import { BaseTransactionBuilder, NotImplementedError, BaseAddress, BaseKey } from 'modules/sdk-core/src';
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-ada/src/transactionBuilder.ts:3]:
```suggestion
import { BaseTransactionBuilder, NotImplementedError, BaseAddress, BaseKey } from '@bitgo/sdk-core';
```

File: modules/sdk-coin-ada/src/transactionBuilderFactory.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import { BaseCoin as CoinConfig } from 'modules/statics/src/base';
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-ada/src/transactionBuilderFactory.ts:1]:
```suggestion
import { BaseCoin as CoinConfig } from '@bitgo/statics';
```

File: modules/sdk-coin-ada/src/transactionBuilderFactory.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+import { BaseTransactionBuilderFactory, NotImplementedError } from 'modules/sdk-core/src';
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-ada/src/transactionBuilderFactory.ts:2]:
```suggestion
import { BaseTransactionBuilderFactory, NotImplementedError } from '@bitgo/sdk-core';
```

File: modules/sdk-coin-ada/package.json [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+      ".ts"
+    ]
+  },
+  "dependencies": {
+    "@bitgo/sdk-core": "^1.0.1",
+    "@bitgo/statics": "^6.17.0",
+    "@bitgo/utxo-lib": "^2.2.2",
+    "bignumber.js": "^9.0.2",
+    "modules": "^0.4.0",
+    "nock": "^13.2.8"
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-ada/package.json:49]: Can delete `nock`. Managed by Lerna at root. See: https://github.com/BitGo/BitGoJS/blob/master/DEVELOPERS.md#dev-dependencies

File: modules/sdk-coin-ada/tsconfig.json [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+  },
+  "include": ["src/**/*", "test"],
+  "exclude": ["node_modules"],
+  "references": [
+    {
+      "path": "../sdk-core"
+    },
+    {
+      "path": "../statics"
+    }
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-ada/tsconfig.json:18]:
```suggestion
    },
    {
    "path": "../utxo-lib"
    }
```

File: yarn.lock [hunk 11]
```diff
@@ -13089,10 +13089,10 @@ parse-json@^5.0.0:
     json-parse-even-better-errors "^2.3.0"
```
Comment (mmcshinsky-bitgo) [inline yarn.lock:13089]: Let's remove the yarn.lock since there aren't any new packages being introduced.

**Review by mmcshinsky-bitgo — CHANGES_REQUESTED** [review]
Let's fix the imports for the packages and we should be good on the skeleton.

File: modules/sdk-coin-ada/src/transactionBuilder.ts (hunk 5 above)
Comment (margueriteblair) [reply modules/sdk-coin-ada/src/transactionBuilder.ts:2]: Also, curious for these imports, why does my local environment throw an error for '@bitgo/statics' packages, but not the above? Wondering if I missed a step in environment setup?

Comment (mmcshinsky-bitgo) [reply modules/sdk-coin-ada/src/transactionBuilder.ts:2]: Hmmm... have you installed the entire repo from the root with `yarn install`?

Comment (margueriteblair) [reply modules/sdk-coin-ada/src/transactionBuilder.ts:2]: Most definitely have done that haha, but let me see if running once more does the trick?

Comment (margueriteblair) [reply modules/sdk-coin-ada/src/transactionBuilder.ts:2]: Of course this time it works -__- but glad it was an easy fix

File: modules/sdk-coin-ada/README.md [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+import { BitGo } from '@bitgo/sdk-api';
+import { BitGoAPI } from '@bitgo/sdk-api';
+import { Ada } from '@bitgo/sdk-coin-ada';
+
+const sdk = new BitGoAPI();
+
+sdk.register('ada', Ada.createInstance);
+
+async function init() {
```
Comment (mmcshinsky-bitgo) [inline modules/sdk-coin-ada/README.md:26]: Can remove the old code here below.

Comment (margueriteblair) [reply modules/sdk-coin-ada/README.md:26]: The entire init() function?

Comment (mmcshinsky-bitgo) [reply modules/sdk-coin-ada/README.md:26]: Yup, everything after sdk.register

**Review by mmcshinsky-bitgo — APPROVED** [review]

**Pull request merged**